src/main/scala/lc/Main.scala

```diff
@@ -56,55 +69,64 @@ class Captcha(throttle: Int, dbConn: DBConn) {
   def getCaptcha(id: Id): Array[Byte] = {
     var image :Array[Byte] = null
     var blob: Blob = null
-    val imageOpt = imagePstmt.synchronized {
+    try {
+      val imagePstmt = Statements.tlStmts.get.imagePstmt
+      val updateTimestampPstmt = Statements.tlStmts.get.updateTimestampPstmt
```

Comment:
IIUC, this will update the timestamp whenever the image for a captcha is requested.

However, the timestamp should be updated only when the challenge is requested, not when the image is requested. Otherwise, if somebody keeps fetching the image, that challenge will never get GCed.

Reply:
Okay. Then we'll have to store another a timestamp field for the last media served to check for the time limit to solve the captcha.